## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 415961`
Commit: `defc9ac6e6d923234115826ea699da719b3b03aa`

---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.brax</li>
    <li>python312Packages.brax.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 31 packages built:</summary>
  <ul>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>mujoco</li>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.dm-control</li>
    <li>python312Packages.dm-control.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.mujoco</li>
    <li>python312Packages.mujoco-mjx</li>
    <li>python312Packages.mujoco-mjx.dist</li>
    <li>python312Packages.mujoco.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.tianshou</li>
    <li>python312Packages.tianshou.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python313Packages.dm-control</li>
    <li>python313Packages.dm-control.dist</li>
    <li>python313Packages.mujoco</li>
    <li>python313Packages.mujoco-mjx</li>
    <li>python313Packages.mujoco-mjx.dist</li>
    <li>python313Packages.mujoco.dist</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.brax</li>
    <li>python312Packages.brax.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 31 packages built:</summary>
  <ul>
    <li>minari (python312Packages.minari)</li>
    <li>minari.dist (python312Packages.minari.dist)</li>
    <li>mujoco</li>
    <li>python312Packages.ale-py</li>
    <li>python312Packages.ale-py.dist</li>
    <li>python312Packages.dm-control</li>
    <li>python312Packages.dm-control.dist</li>
    <li>python312Packages.gymnasium</li>
    <li>python312Packages.gymnasium.dist</li>
    <li>python312Packages.mujoco</li>
    <li>python312Packages.mujoco-mjx</li>
    <li>python312Packages.mujoco-mjx.dist</li>
    <li>python312Packages.mujoco.dist</li>
    <li>python312Packages.pettingzoo</li>
    <li>python312Packages.pettingzoo.dist</li>
    <li>python312Packages.shimmy</li>
    <li>python312Packages.shimmy.dist</li>
    <li>python312Packages.skrl</li>
    <li>python312Packages.skrl.dist</li>
    <li>python312Packages.stable-baselines3</li>
    <li>python312Packages.stable-baselines3.dist</li>
    <li>python312Packages.tianshou</li>
    <li>python312Packages.tianshou.dist</li>
    <li>python312Packages.torchrl</li>
    <li>python312Packages.torchrl.dist</li>
    <li>python313Packages.dm-control</li>
    <li>python313Packages.dm-control.dist</li>
    <li>python313Packages.mujoco</li>
    <li>python313Packages.mujoco-mjx</li>
    <li>python313Packages.mujoco-mjx.dist</li>
    <li>python313Packages.mujoco.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python312Packages.brax</summary>
<pre>brax/io/json_test.py::JsonTest::test_dumps
  /nix/store/2qfwxvhrapjhzza19gwqpmilhm44wgig-python3.12-jax-0.6.1/lib/python3.12/site-packages/jax/_src/interpreters/xla.py:119: RuntimeWarning: overflow encountered in cast
    return np.asarray(x, dtypes.canonicalize_dtype(x.dtype))

brax/v1/tests/colliders_test.py: 6 warnings
brax/v1/tests/physics_test.py: 127 warnings
brax/v1/tests/env_test.py: 1 warning
  /build/source/brax/v1/physics/base.py:197: UserWarning: dynamics_mode either not specified or not recognized, defaulting to "pbd".  If you wish to preserve legacy behavior used in previous versions of Brax, set dynamics_mode="legacy_spring".
    warnings.warn(

brax/training/agents/ars/train_test.py::ARSTest::testModelEncoding0
  /nix/store/2qfwxvhrapjhzza19gwqpmilhm44wgig-python3.12-jax-0.6.1/lib/python3.12/site-packages/jax/_src/numpy/reductions.py:1250: DeprecationWarning: jnp.std: where must be None or a boolean array; got dtype=int32.
    where = check_where("std", where)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED brax/mjx/perf_test.py::PerfTest::test_pipeline_ant - TypeError: Data.__init__() got an unexpected keyword argument 'contact'
FAILED brax/mjx/pipeline_test.py::PipelineTest::test_pendulum - TypeError: Data.__init__() got an unexpected keyword argument 'contact'
FAILED brax/mjx/pipeline_test.py::PipelineTest::test_pipeline_init_with_ctrl - TypeError: Data.__init__() got an unexpected keyword argument 'contact'
=========== 3 failed, 894 passed, 411 warnings in 860.93s (0:14:20) ============</pre>
</details>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.brax</summary>
<pre>
brax/v1/experimental/composer/tests/composer_test.py::ComposerTest::testEnvCreation0
brax/v1/experimental/composer/tests/composer_test.py::ComposerTest::testMultiAgentEnvCreation0
brax/v1/experimental/composer/tests/composer_test.py::ComposerTest::testMultiAgentEnvCreation1
brax/v1/experimental/composer/tests/composer_test.py::ComposerTest::testEnvCreation1
brax/v1/experimental/composer/tests/composer_test.py::ComposerTest::testEnvCreation2
  /build/source/brax/v1/experimental/composer/composer.py:178: UserWarning: Explicitly requested dtype float requested in array is not available, and will be truncated to dtype float32. To enable more dtypes, set the jax_enable_x64 configuration option or the JAX_ENABLE_X64 shell environment variable. See https://github.com/jax-ml/jax#current-gotchas for more.
    new_v['quat'] = jnp.array(

brax/contact_test.py::ConvexTest::test_convex_convex
brax/io/json_test.py::JsonTest::test_dumps
  /nix/store/l6h5x41dfhqwr35z1ip2602s6k7z0ddq-python3.12-jax-0.6.1/lib/python3.12/site-packages/jax/_src/interpreters/xla.py:119: RuntimeWarning: overflow encountered in cast
    return np.asarray(x, dtypes.canonicalize_dtype(x.dtype))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED brax/mjx/perf_test.py::PerfTest::test_pipeline_ant - TypeError: Data.__init__() got an unexpected keyword argument 'contact'
FAILED brax/mjx/pipeline_test.py::PipelineTest::test_pendulum - TypeError: Data.__init__() got an unexpected keyword argument 'contact'
FAILED brax/mjx/pipeline_test.py::PipelineTest::test_pipeline_init_with_ctrl - TypeError: Data.__init__() got an unexpected keyword argument 'contact'
=========== 3 failed, 891 passed, 496 warnings in 226.65s (0:03:46) ============</pre>
</details>
</details>

--- a/pkgs/development/python-modules/brax/default.nix
+++ b/pkgs/development/python-modules/brax/default.nix
@@ -89,11 +89,20 @@ buildPythonPackage rec {
     transforms3d
   ];
 
-  disabledTests = lib.optionals stdenv.hostPlatform.isAarch64 [
-    # Flaky:
-    # AssertionError: Array(-0.00135638, dtype=float32) != 0.0 within 0.001 delta (Array(0.00135638, dtype=float32) difference)
-    "test_pendulum_period2"
-  ];
+  disabledTests =
+    [
+      # Broken after mujoco was updated to 3.3.3:
+      # TypeError: Data.init() got an unexpected keyword argument 'contact'
+      # Reported upstream: https://github.com/google/brax/issues/618
+      "test_pendulum"
+      "test_pipeline_ant"
+      "test_pipeline_init_with_ctrl"
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isAarch64 [
+      # Flaky:
+      # AssertionError: Array(-0.00135638, dtype=float32) != 0.0 within 0.001 delta (Array(0.00135638, dtype=float32) difference)
+      "test_pendulum_period2"
+    ];
 
   disabledTestPaths = [
     # ValueError: matmul: Input operand 1 has a mismatch in its core dimension

